java/src/main/java/com/powsybl/python/network/NetworkModificationCFunctions.java

```diff
@@ -0,0 +1,150 @@
+    public static PyPowsyblApiHeader.ArrayPointer<PyPowsyblApiHeader.SeriesPointer> getConnectablesOrderPositions(IsolateThread thread, ObjectHandle networkHandle,
+                                                     CCharPointer voltageLevelId, PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            String voltageLevelIdStr = CTypeUtil.toString(voltageLevelId);
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+
+            VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelIdStr);
+            Map<String, List<ConnectablePosition.Feeder>> feederPositionsOrders = getFeedersByConnectable(voltageLevel);
+            return Dataframes.createCDataframe(Dataframes.feederMapMapper(), feederPositionsOrders);
+        });
+
+    }
+
+    @CEntryPoint(name = "getUnusedConnectableOrderPositionsBefore")
+    public static PyPowsyblApiHeader.ArrayPointer<CIntPointer> getUnusedConnectableOrderPositionsBefore(IsolateThread thread, ObjectHandle networkHandle,
+                                                                                             CCharPointer busbarSectionId,
+                                                                                             PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        Network network = ObjectHandles.getGlobal().get(networkHandle);
+        String busbarSectionIdStr = CTypeUtil.toString(busbarSectionId);
+        BusbarSection busbarSection = network.getBusbarSection(busbarSectionIdStr);
+        try {
+            Optional<Range<Integer>> positionsOrders = getUnusedOrderPositionsBefore(busbarSection);
+            if (positionsOrders.isPresent()) {
+                int max = positionsOrders.get().getMaximum();
+                int min = positionsOrders.get().getMinimum();
+                return createIntegerArray(Arrays.asList(min, max));
+            } else {
+                return createIntegerArray(Collections.EMPTY_LIST);
+            }
+        } catch (PowsyblException e) {
+            throw new PowsyblException(e.getMessage());
+        }
+    }
+
+    @CEntryPoint(name = "getUnusedConnectableOrderPositionsAfter")
+    public static PyPowsyblApiHeader.ArrayPointer<CIntPointer> getUnusedConnectableOrderPositionsAfter(IsolateThread thread, ObjectHandle networkHandle,
+                                                                                                          CCharPointer busbarSectionId,
+                                                                                                          PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        Network network = ObjectHandles.getGlobal().get(networkHandle);
+        String busbarSectionIdStr = CTypeUtil.toString(busbarSectionId);
+        BusbarSection busbarSection = network.getBusbarSection(busbarSectionIdStr);
+        try {
```

Comment:
(if we keep the method)
The try catch is not useful, but we need to wrap the method in a `doCatch` to propagate exceptions to C code.